apps/src/netsim/NetSimTable.js

```diff
@@ -230,18 +269,43 @@ NetSimTable.prototype.synchronousDelete = function (id) {
  */
 NetSimTable.prototype.fullCacheUpdate_ = function (allRows) {
   // Rebuild entire cache
+  var maxRowID = 0;
   var newCache = allRows.reduce(function (prev, currentRow) {
     prev[currentRow.id] = currentRow;
+    if (currentRow.id > maxRowID) {
+      maxRowID = currentRow.id;
+    }
     return prev;
   }, {});
 
   // Check for changes, if anything changed notify all observers on table.
   if (!_.isEqual(this.cache_, newCache)) {
     this.cache_ = newCache;
+    this.latestRowID_ = maxRowID;
+    this.tableChange.notifyObservers(this.arrayFromCache_());
+  }
+
+  this.lastRefreshTime_ = Date.now();
+};
+
+/**
+ * @param {Array} newRows
+ * @private
+ */
+NetSimTable.prototype.incrementalCacheUpdate_ = function (newRows) {
+  if (newRows.length > 0) {
+    var maxRowID = 0;
+    newRows.forEach(function (row) {
+      this.cache_[row.id] = row;
+      if (row.id > maxRowID) {
+        maxRowID = row.id;
+      }
+    }, this);
+    this.latestRowID_ = maxRowID;
```

Comment:
This code doesn't do the optimization of only notifying observers if there are changes.  I believe it should be possible to refactor the code so that most of the code is shared with fullCacheUpdate_ to ensure consistent behavior in both cases.